src/rgw/rgw_common.cc

```diff
@@ -70,6 +84,169 @@ void rgw_perf_stop(CephContext *cct)
   delete perfcounter;
 }
 
+
+int percentile_perf_start(CephContext *cct)
+{
+  int p_starti, p_endi, p_inci, p_updatei, p_reseti, index = 1;
+  percentile_get_first = 16000;
+
+  vector<string> sections;
+  sections.push_back("percentile");
+  string percentiles_conf_val, object_sizes_conf_val, p_start, p_end, p_inc, p_update, p_reset;
+  cct->_conf->get_val_from_conf_file(sections, "percentiles", percentiles_conf_val, false);
+  cct->_conf->get_val_from_conf_file(sections, "object_sizes", object_sizes_conf_val, false);
+  cct->_conf->get_val_from_conf_file(sections, "start", p_start, false);
+  cct->_conf->get_val_from_conf_file(sections, "end", p_end, false);
+  cct->_conf->get_val_from_conf_file(sections, "inc", p_inc, false);
+  cct->_conf->get_val_from_conf_file(sections, "update", p_update, false);
+  cct->_conf->get_val_from_conf_file(sections, "reset", p_reset, false);
+
+  stringstream ss;
+  string item;
+
+  if(percentiles_conf_val.size() > 0) {
+    ss.str(percentiles_conf_val);
+    while (getline(ss, item, ',')) {
+      percentiles.push_back(std::atof(item.c_str()));
+    }
+  }
+  if(object_sizes_conf_val.size() > 0) {
+    ss.clear();
+    ss.str("");
+    item = "";
+    ss.str(object_sizes_conf_val);
+    while (getline(ss, item, ',')) {
+      object_sizes.push_back(std::atoi(item.c_str()));
+    }
+  }
+
+  if(percentiles.empty()) {
+    percentiles.push_back(0.99);
```

Comment:
Please don't hardcode this, include these as default values for the config options.